File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/3372-longest-strictly-increasing-or-strictly-decreasing-subarray.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        int max = 1, inc_sub = 1, dec_sub = 1;

        for(int i = 1; i < nums.length; i++){
            if(nums[i] < nums[i - 1]){
                dec_sub += 1;
                inc_sub = 1; 
            }
            else if(nums[i] > nums[i - 1]){
                inc_sub += 1;
                dec_sub = 1;
            }
            else{
                inc_sub = 1;
                dec_sub = 1;
            }
            max = Math.max(max, Math.max(inc_sub, dec_sub));
        }
        return max;
    }
    
}

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/3372-longest-strictly-increasing-or-strictly-decreasing-subarray.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int incre = 1, decre = 1, max_length = 1;;
        for(int i = 0; i < nums.size() - 1; i++){
            if(nums[i] < nums[i + 1]){
                incre += 1;
                decre = 1;
            }
            else if(nums[i] > nums[i + 1]){
                decre += 1;
                incre = 1;
            }
            else{
                decre = incre = 1;
            }
            max_length = max(max_length, max(incre, decre));
        }
        return max_length;
    }
};